From 893086366f8ab2e0dc4af5452cc3a7dbd92281a1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Wed, 3 Jan 2024 04:41:57 +0100
Subject: [PATCH] Use cmake_path over get_filename_component

get_filename_component has been superseded by cmake_path in CMake 3.20.
---
 CMakeLists.txt     | 4 ++--
 cmake/Macros.cmake | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9e084b2..aa5f1dc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -185,11 +185,11 @@ if(ENABLE_BANKDATA_DOWNLOAD)
 else()
     # to be removed
     file(GLOB BANKDATA_FILEPATH  "${BANKDATA_SRCPATH}/bankdata_*.txt")
-    get_filename_component(BANKDATA_FILE ${BANKDATA_FILEPATH} NAME)
+    cmake_path(GET BANKDATA_FILEPATH FILENAME BANKDATA_FILE)
     message(STATUS "Using present bank data file ${BANKDATA_FILEPATH}")
 
     file(GLOB BANKDATA_RAW_FILEPATH  "${BANKDATA_SRCPATH}/blz_*.txt")
-    get_filename_component(BANKDATA_RAW_FILE ${BANKDATA_RAW_FILEPATH} NAME)
+    cmake_path(GET BANKDATA_RAW_FILEPATH FILENAME BANKDATA_RAW_FILE)
     message(STATUS "Using present raw bank data file ${BANKDATA_RAW_FILEPATH}")
 endif()
 set(BANKDATA_TEST_FILEPATH "${BANKDATA_SRCPATH}/bankdata_test.txt")
diff --git a/cmake/Macros.cmake b/cmake/Macros.cmake
index 6fb0f55..74aa388 100644
--- a/cmake/Macros.cmake
+++ b/cmake/Macros.cmake
@@ -17,7 +17,7 @@ macro(setup_filenames)
         message(STATUS "Using data file ${DATA_FILEPATH} for key ${__KEY}")
     else()
         file(GLOB DATA_FILEPATH "${CMAKE_CURRENT_SOURCE_DIR}/${__PREFIX}_*.txt")
-        get_filename_component(DATA_FILE ${DATA_FILEPATH} NAME)
+        cmake_path(GET DATA_FILEPATH FILENAME DATA_FILE)
         message(STATUS "Using present data file '${DATA_FILEPATH}' for key ${__KEY}")
     endif()
     set(DATA_FILEPATH_${__KEY} ${DATA_FILEPATH}  CACHE STRING "raw data file path" FORCE)
